modules/install/wizard.php

```diff
@@ -199,14 +199,72 @@
 
             if ($res == 1 or $res == 3 or $res == 5) {
                 $db->connect();
+                
+            if ($db->success) {
+                
+                $dsp->AddFieldSetStart(t('MySQL'));
+
+                // SQL mode - check for options with known issues
+                $sqlmode_disable = array('ONLY_FULL_GROUP_BY');
+                $res = $db->qry('SELECT @@SESSION.SQL_MODE as sqlmode;');
+                $opts = $db->fetch_array($res)['sqlmode'];
+                $server_opts = explode (',',$opts);
+                $warn_opts = implode(array_values(array_intersect($server_opts,$sqlmode_disable)),',');
+                $sql_mode='';
+                if ($warn_opts) {
+                    $sql_mode = $warning . t("Der MySQL-Server hat aktuell folgende Optionen aktiviert, die zu Problemen mit LanSuite führen: %1 \nEs wird versucht, dies automatisch zu kompensieren, indem diese für jede Verbindung deaktiviert werden. Dies hat zur Folge, dass eventuelle Änderungen an der Server-Variable SQL_MODE nicht automatisch für LanSuite übernommen werden, sondern manuell in config.php angepasst werden müssen", $warn_opts);
+                    $config['database']['sqlmode']=implode(',',array_diff($server_opts,$sqlmode_disable));
+                    $install->WriteConfig();
+                    
+                } else {
+                    $sql_mode = $ok . t("Folgende SQL-Optionen werden vom Server für die Verbindung gesetzt: %1",$opts);
+                }
```

Comment:
The else was there so that the SQL-mode is not reported as OK, but as Warning.
By doing per suggestion it would be first reported as OK, then a warning for it would be filled out.

Not sure that is an improvement? ;)

Reply:
The `$sql_mode` variable is getting overwritten before being added as a row to the output. Thats why it is not getting reported twice, but only once.
The improvement here is skipping the else part at all and setting the OK part as default. Only when there is a warning the OK get overwritten.

This means one control flow less, which is usually, easier to read/understand.

Reply:
Ah, you are right, sorry!